lib/broadway/topology/rate_limiter.ex

```diff
@@ -41,21 +39,25 @@ defmodule Broadway.Topology.RateLimiter do
     GenServer.call(rate_limiter, :get_rate_limiting)
   end
 
+  def get_rate_limiter_ref(rate_limiter) do
+    GenServer.call(rate_limiter, :get_rate_limiter_ref)
+  end
+
   @impl true
-  def init({broadway_name, rate_limiting_opts, producers_names}) do
+  def init({_broadway_name, rate_limiting_opts, producers_names}) do
     interval = Keyword.fetch!(rate_limiting_opts, :interval)
     allowed = Keyword.fetch!(rate_limiting_opts, :allowed_messages)
 
-    table = :ets.new(table_name(broadway_name), [:named_table, :public, :set])
-    :ets.insert(table, {@row_name, allowed})
+    counter = :atomics.new(1, [])
+    :atomics.put(counter, 1, allowed)
```

Comment:
With my suggestion above, this would become

```suggestion
    :atomics.put(counter, @atomics_index, allowed)
```